core/src/main/scala/org/apache/gearpump/cluster/Application.scala

```diff
@@ -29,8 +29,53 @@ case class AppJar(name: String, container: JarFileContainer)
  * Subtype this for user defined application
  */
 trait Application {
+  val name : String
   val appMaster : String
-  val conf: Configs
+  val conf: UserConfig
+}
+
+/**
+ * Used for verification
+ */
+abstract class ApplicationMaster extends Actor
+
+/**
+ * Used for verification
+ */
+abstract class ApplicationExecutor extends Actor
+
+
+/**
+ * This interface should keep stable, so that we can remain
+ * compatible with old versions
+ */
+trait ExecutorContextInterface {
+  val executorId : Int
+  val workerId : Int
+  val appId : Int
+  val appMaster : ActorRef
+  val startClock : TimeStamp
```

Comment:
I think startClock field is mainly related to the streaming type application, so is it necessary for this general interface?